### ISSUE TITLE
but buffered writes should be faster

### ISSUE THREAD

**Issue by blackmius** [Alogani/asyncio#2]
https://github.com/Alogani/asyncio/blob/9320a3e469a69b1e53e4934b547e295e5a74f4fa/src/asyncio/private/asyncbuffer.nim#L37

i did benchmarks on what size and how often writes should be done and writing rarely and bigger chunks showing bigger throughput, this also true for networking especially of you apply some sort of compression before write

**Comment by Alogani** [comment]
Hi,

I focused on files and pipes on the system, so I didn't put much attention into it. For files and unix pipes, it was really slow on read and write combined (for ssd at least). But you are right, its a nice addition, and have its utility.

I don't know if I can work on it soon, but I will create branch for this issue. (And i you want to contribute, you are welcome !)

I have wrote the object signatures I think are the more approriate here : https://github.com/Alogani/asyncio/blob/2-implement-asyncbuffer/src/asyncio/asyncbuffer.nim

Thanks for your interest !
